app/src/main/kotlin/com/wire/android/WireApplication.kt

```diff
@@ -153,21 +159,27 @@ class WireApplication : BaseApp() {
         initializeAnonymousAnalytics()
     }
 
-    private fun initializeAnonymousAnalytics() {
+    private suspend fun initializeAnonymousAnalytics() {
         val anonymousAnalyticsRecorder = AnonymousAnalyticsRecorderImpl()
         val analyticsSettings = AnalyticsSettings(
             countlyAppKey = BuildConfig.ANALYTICS_APP_KEY,
             countlyServerUrl = BuildConfig.ANALYTICS_SERVER_URL,
             enableDebugLogging = BuildConfig.DEBUG
         )
 
-        AnonymousAnalyticsManagerImpl.init(
-            context = this,
-            analyticsSettings = analyticsSettings,
-            isEnabledFlowProvider = globalDataStore.get()::isAnonymousUsageDataEnabled,
-            anonymousAnalyticsRecorder = anonymousAnalyticsRecorder,
-            dispatcher = Dispatchers.IO
-        )
+        coreLogic.get().getGlobalScope().session.currentSessionFlow().collectLatest { sessionResult ->
+            if (sessionResult is CurrentSessionResult.Success && sessionResult.accountInfo.isValid()) {
+                val userDataStore = userDataStoreProvider.get().getOrCreate(sessionResult.accountInfo.userId)
+
+                AnonymousAnalyticsManagerImpl.init(
+                    context = this,
+                    analyticsSettings = analyticsSettings,
+                    isEnabledFlowProvider = userDataStore::isAnonymousUsageDataEnabled,
+                    anonymousAnalyticsRecorder = anonymousAnalyticsRecorder,
+                    dispatcher = Dispatchers.IO
+                )
+            }
+        }
```

Comment:
does this need the check if if the build have analytics enabled?

Reply:
before the check was inside AnonymousAnalyticsManagerImpl.init but now since we are doing some extra work with the user session maybe we can also check for it here

Reply:
The verification is done in gradle level, where if its enable we include whatever analytics we have, otherwise we will be using stub calls here, so `AnonymousAnalyticsManagerImpl.init(..)` will call "nothing" and return an `Unit`

Does this makes sense? as its done in gradle including libs, or shall we do another check here?

Reply:
i mean to now run the whole check for current active user in case it is not enabled for the build

Reply:
Added the check for analytics enabled from build config